src/change_stream.ts

```diff
@@ -568,25 +578,50 @@ function setIsIterator<TSchema>(changeStream: ChangeStream<TSchema>): void {
   }
   changeStream[kMode] = 'iterator';
 }
+
+function createChangeStreamStageOptions<TSchema>(
+  changeStream: ChangeStream<TSchema>,
+  changeStreamOptions: ChangeStreamOptions
+) {
+  const changeStreamStageOptions: Document = {
+    fullDocument: changeStreamOptions.fullDocument
+  };
+
+  for (const optionName of CHANGE_STREAM_OPTIONS) {
+    if (changeStreamOptions[optionName]) {
+      changeStreamStageOptions[optionName] = changeStreamOptions[optionName];
+    }
+  }
+
+  if (changeStream.type === CHANGE_DOMAIN_TYPES.CLUSTER) {
+    changeStreamStageOptions.allChangesForCluster = true;
+  }
+
+  return changeStreamStageOptions;
+}
 /**
  * Create a new change stream cursor based on self's configuration
  * @internal
  */
 function createChangeStreamCursor<TSchema>(
   changeStream: ChangeStream<TSchema>,
-  options: ChangeStreamOptions
+  changeStreamOptions: ChangeStreamOptions
 ): ChangeStreamCursor<TSchema> {
-  const changeStreamStageOptions: Document = { fullDocument: options.fullDocument || 'default' };
-  applyKnownOptions(changeStreamStageOptions, options, CHANGE_STREAM_OPTIONS);
-  if (changeStream.type === CHANGE_DOMAIN_TYPES.CLUSTER) {
-    changeStreamStageOptions.allChangesForCluster = true;
-  }
-
+  const changeStreamStageOptions = createChangeStreamStageOptions(
+    changeStream,
+    changeStreamOptions
+  );
```

Comment:
```suggestion
  const changeStreamStageOptions = {
    resumeAfter: options.resumeAfter != null ? options.resumeAfter : undefined,
    startAfter: Boolean(options.startAfter),
    startAtOperationTime:
      options.startAtOperationTime != null ? options.startAtOperationTime : undefined,
    fullDocument: options.fullDocument != null ? options.fullDocument : undefined,
    allChangesForCluster: changeStream.type === CHANGE_DOMAIN_TYPES.CLUSTER
  };
  const cursorOptions = {
    ...changeStreamStageOptions,
    batchSize: options.batchSize != null ? options.batchSize : undefined,
    maxAwaitTimeMS: options.maxAwaitTimeMS != null ? options.maxAwaitTimeMS : undefined,
    collation: options.collation != null ? options.collation : undefined,
    readPreference: options.readPreference != null ? options.readPreference : undefined
  };
```

I was thinking more very bespoke, and delete the list of options at the top of this file, doing things this way is verbose but it catches more errors. When I did this I found we have a small bug in the `get resumeOptions` logic

Reply:
Per our conversation on slack, I re-added applyKnownOptions.